addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -175,9 +175,9 @@ def _get_free_shipping_lines(self):
         return self.order_line.filtered(lambda l: l.reward_id.reward_type == 'shipping')
 
     def _allow_nominative_programs(self):
-        if not request or not hasattr(request, 'website'):
-            return super()._allow_nominative_programs()
-        return not request.website.is_public_user() and super()._allow_nominative_programs()
+        if request and hasattr(request, 'website') and request.website.is_public_user():
+            return self.state == 'sent' and super()._allow_nominative_programs()
```

Comment:
Can't we sign confirm orders?

Reply:
We can pay confirmed orders, but paying confirmed orders will not affect loyalty. Loyalty is done upon confirmation.